### ISSUE TITLE
Export UploadedFile scalar resolver

### ISSUE THREAD

**Issue by jaydenseric** [HriBB/graphql-server-express-upload#9]
Step 3 in the instructions, "[Add UploadedFile resolver](https://github.com/HriBB/graphql-server-express-upload#3-add-uploadedfile-resolver)" is too hard to follow.

Pasting the example code into a typical Apollo resolvers.js file wont work because `Kind` is undefined. Where does that come from, how do you import it? And what is this `ast` parameter?

This looks pretty boilerplate, why not [export the scalar resolver](http://dev.apollodata.com/tools/graphql-tools/scalars.html#Using-a-package) for easy setup?

Something like:

```js
import {UploadedFileResolver} from 'graphql-server-express-upload'

export default {
  UploadedFile: UploadedFileResolver,
  Query: { ... },
  Mutation: {  ... }
}
```

In the meantime, I can't work out how to get the resolver to work so any clarification would be super helpful.

**Comment by jaydenseric** [comment]
I got this working by dropping in [taion/graphql-type-json](https://github.com/taion/graphql-type-json). [The name is `JSON`](https://github.com/taion/graphql-type-json/blob/v0.1.4/src/index.js#L32), when ideally it would be `UploadedFile`, but it seems to be working ok.

```js
import GraphQLJSON from 'graphql-type-json'
export default {
  UploadedFile: GraphQLJSON,
  Query: { ... },
  Mutation: {  ... }
}
```

**Comment by HriBB** [comment]
@jaydenseric I agree. Do you want to make a PR?

**Comment by jaydenseric** [comment]
Perhaps, I am in the middle of some epic refactoring but might be in a place to look at it sometime in the next few days.

The importable resolver might be better placed in [HriBB/apollo-upload-network-interface](https://github.com/HriBB/apollo-upload-network-interface) so it does not have to be duplicated for future non-express middleware. Thoughts?

**Comment by HriBB** [comment]
Yeah I am very busy ATM as well ... will try to update all packages in a batch sometimes soon.

I think that scalar should be a part of the server package(s), because it is not needed on the client. Or it could be a part of some external package or even its own package. I was thinking about using [lerna](https://github.com/lerna/lerna) for package management and typescript. Check out [this issue](https://github.com/HriBB/apollo-upload-network-interface/issues/6)

**Comment by jaydenseric** [comment]
A benefit to keeping it here is the `graphql` dependency; it already exists here but not yet in apollo-upload-network-interface.

I for one can't stand TypeScript in OSS, it's not real JavaScript and it makes contributing or even understanding the source difficult. [A lot of traffic seems to be about fiddling around with types](https://github.com/apollostack/apollo-client/search?q=type&type=Commits&utf8=%E2%9C%93).

**Comment by vladinator1000** [comment]
I would love to find out more about this. I've been trying to upload a file for a couple of weeks but no luck.

**Comment by jaydenseric** [comment]
@savovs If you're still stuck, you might find [apollo-upload-server](https://github.com/jaydenseric/apollo-upload-server) a little easier to get up and running – you don't have to set up a custom scalar or Multer.
